vars/needsRebuild.groovy

```diff
@@ -0,0 +1,21 @@
+def call(archiveDir, commit, lvVersions) {
+   // We can't source a Python virtualenv in one call and use it in another,
+   // because Groovy will spawn a separate shell for each call.
+   // Instead, build a multi-line string and execute it all at once.
+   def venvDir = "env"
+   def trimmedDir = archiveDir.trim()
+   def resourcesDir = "${WORKSPACE}\\niveristand-custom-device-build-tools\\resources"
+   def output = bat returnStdout: true, script: """
```

Comment:
I'd love to find a way to avoid duplicating this boilerplate, and potentially re-use the existing virtual environment to avoid continually recreating them.

I don't think that work has to be in scope here, but if you have ideas I'd love to hear them.

Reply:
I don't like it either. I thought about potentially having a virtualEnv.groovy that has all of this stuff and takes a string for the actual python command, but didn't implement it for this. I might experiment a little.

Reply:
I won't hold the review on it, because there's precedence of us duplicating this logic already.